### PR TITLE
Scoreboard UI

### PR DESCRIPTION
**Pull request by JeremyDezalos**

### CODE REVIEW

File: app/src/main/res/layout/scoreboard_entry.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="59dp"
+        android:layout_margin="4dp">
+
+        <TextView
+            android:id="@+id/scoreboardUsername"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_alignParentTop="true"
+            android:text="username"
```
Comment (rkuenzi-epfl) [inline app/src/main/res/layout/scoreboard_entry.xml:19]: Lint warning:
"Hardcoded string "username", should use `@string` resource"

File: app/src/main/res/layout/activity_scoreboard.xml [hunk 2]
```diff
@@ -16,10 +16,10 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent">
 
-        <TableLayout
-            android:id="@+id/scoreTable"
+        <androidx.recyclerview.widget.RecyclerView
             android:layout_width="match_parent"
-            android:layout_height="wrap_content" />
+            android:id="@+id/recyclerViewScoreboard"
+            android:layout_height="match_parent"/>
```
Comment (rkuenzi-epfl) [inline app/src/main/res/layout/activity_scoreboard.xml:22]: Lint warning:
"This androidx.recyclerview.widget.RecyclerView should use `android:layout_height="wrap_content"`"
It seems he is unhappy that you use "match_parent" but I guess the result will look a bit different if you change it?

**Review by rkuenzi-epfl — CHANGES_REQUESTED** [review]
There's some lint warning but otherwise it's great.
Also I think we can find a better button, maybe a floating action button with a <- that we could put on the bottom right over the scoreboard.
The scores could also use some improvement, maybe a slightly bigger text for the values and eventually write the place on the left.

File: app/src/main/java/com/github/wnder/ScoreboardActivity.java [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.lifecycle.ViewModelProvider;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
 
 import android.os.Bundle;
 import android.widget.TableLayout;
```
[the remark this answers is not in the document]
Comment (JeremyDezalos) [reply app/src/main/java/com/github/wnder/ScoreboardActivity.java]: Done

**Review by JeremyDezalos — COMMENTED** [review]
Definitely the best way to answer comments

**Review by rkuenzi-epfl — APPROVED** [review]

File: app/src/main/res/layout/activity_scoreboard.xml [hunk 4]
```diff
@@ -4,31 +4,48 @@
...
         android:layout_height="wrap_content"
-        android:text="@string/leave_scoreboard"
+        android:layout_marginEnd="8dp"
+        android:layout_marginBottom="32dp"
+        android:clickable="true"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintHorizontal_bias="0.5"
-        app:layout_constraintStart_toStartOf="parent" />
+        app:srcCompat="@drawable/ic_baseline_arrow_back_24"
```
Comment (amattei777) [inline app/src/main/res/layout/activity_scoreboard.xml:48]: We will have one day to discuss with the team to decide on some guidelines for the typical margins/colors to use in the app so we use all the same. We will do that later apparently thought.

**Review by amattei777 — APPROVED** [review]
Didn't find any major problem. We will discuss to uniformize the details that are left for the design during the last stand up meeting I believe

**Review by amattei777 — APPROVED** [review]

**Pull request merged**